### PR TITLE
Complements the documentation for pattern files and exclude files

### LINKED ISSUE #5236: Example path for --exclude-from
## Have you checked borgbackup docs, FAQ, and open Github issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

ISSUE

## Describe the problem you're observing.

I found it quite hard to find out how to use/allow spaces in --exclude-from file, one extra example in manual/help might help a lot:

# No need to escape spaces 
/mnt/c/System Volume Information

(not having to escape spaces is quite counterintuitive :)

### PR DESCRIPTION
**Pull request by Gu1nness**
Fixes #5236

### CODE REVIEW

File: src/borg/archiver.py [hunk 1]
```diff
@@ -2644,11 +2644,13 @@ def define_exclude_and_patterns(add_option, *, tag_files=False, strip_components
...
             add_option('--exclude-from', metavar='EXCLUDEFILE', action=ArgparseExcludeFileAction,
-                       help='read exclude patterns from EXCLUDEFILE, one per line')
+                       help='read exclude patterns from EXCLUDEFILE, one per'
+                       'line. These entries are processed by borg and not the shell')
             add_option('--pattern', metavar='PATTERN', action=ArgparsePatternAction,
                        help='experimental: include/exclude paths matching PATTERN')
             add_option('--patterns-from', metavar='PATTERNFILE', action=ArgparsePatternFileAction,
-                       help='experimental: read include/exclude patterns from PATTERNFILE, one per line')
+                       help='experimental: read include/exclude patterns from PATTERNFILE, one per line'
+                       'These entries are processed by borg and not the shell')
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:2653]: maybe this is a) too long fo here and b) not really helpful as the user would need to be already aware of what the specialities of shell processing are.

so guess we better add this to the (long) docs and also give an example there that shows that spaces are not escaped in pattern files.

Comment (Gu1nness) [reply src/borg/archiver.py:2653]: Sure I'll do this now.

File: src/borg/archiver.py [hunk 2]
```diff
@@ -3106,6 +3106,18 @@ def define_borg_mount(parser):
...
+        $ cat exclude-file.txt
+        # Comment line
+        /home/*/junk
+        *.tmp
+        One file with spaces
+        ```
+        Then the command borg ``create --exclude-from exclude-file.txt`` will
+        delete all files excluding /home/*/junk, *.tmp and the file named
+        ``One file with spaces``
+
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:3120]: be careful when using markup in archiver.py:

while the rst / sphinx docs (-> online, html, pdf) support rather powerful markup, the stuff that `borg help` outputs should also look ok (and guess we only support a rather limited subset of the markup for that).

for each command, there is also a separate .rst.inc (.inc.rst?) file with examples, that could also be used.

one could also check that "borg help patterns" text, considering it is already mentioned a line above and that we need to refer to such an example for multiple borg commands, at least create, recreate and mount - so putting it just into the create help is maybe not the best solution and also causes duplication.

Comment (Gu1nness) [reply src/borg/archiver.py:3120]: I'll put it in the borg help patterns then :)

File: src/borg/archiver.py [hunk 3]
```diff
@@ -2228,6 +2228,8 @@ def do_break_lock(self, args, repository):
             re:^home/[^/]\\.tmp/
             sh:home/*/.thumbnails
             EOF
+            # Example with spaces, no need to escape as it is processed by borg
+            some file with spaces.txt
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:2232]: you need to put that above the EOF line so it becomes part of that file.

File: src/borg/archiver.py [hunk 4]
```diff
@@ -2644,7 +2646,7 @@ def define_exclude_and_patterns(add_option, *, tag_files=False, strip_components
                        type=parse_exclude_pattern, action='append',
                        help='exclude paths matching PATTERN')
             add_option('--exclude-from', metavar='EXCLUDEFILE', action=ArgparseExcludeFileAction,
-                       help='read exclude patterns from EXCLUDEFILE, one per line')
+                       help='read exclude patterns from EXCLUDEFILE, one per line.')
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:2649]: please remove accidental change.

File: src/borg/archiver.py [hunk 5]
```diff
@@ -3605,8 +3619,6 @@ def define_borg_mount(parser):
         can be selected by passing a list of ``PATHs`` as arguments.
         The file selection can further be restricted by using the ``--exclude`` option.
 
-        See the output of the "borg help patterns" command for more help on exclude patterns.
-
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:3609]: accidental change?

File: src/borg/archiver.py [hunk 6]
```diff
@@ -3637,7 +3649,6 @@ def define_borg_mount(parser):
         be restricted by using the ``--exclude`` option.
 
         See the output of the "borg help patterns" command for more help on exclude patterns.
-
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:3640]: accidental change?

File: src/borg/archiver.py [hunk 7]
```diff
@@ -3980,8 +3991,6 @@ def define_borg_mount(parser):
         list_epilog = process_epilog("""
         This command lists the contents of a repository or an archive.
 
-        See the "borg help patterns" command for more help on exclude patterns.
-
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:3984]: accidental change?

**Review by ThomasWaldmann — COMMENTED** [review]
would be good if you (in general) looked at the overall diff on github, so you'll notice (and fix) accidental changes.

**Comment by Gu1nness** [comment]
Sorry for all the accidental removals, I checked on my laptop but it did not appear to my eyes that these lines had disappeared…

File: src/borg/archiver.py (hunk 4 above)
Comment (ThomasWaldmann) [reply src/borg/archiver.py:2649]: todo

File: src/borg/archiver.py (hunk 2 above)
Comment (ThomasWaldmann) [reply src/borg/archiver.py:3122]: is this still needed?

**Comment by ThomasWaldmann** [comment]
Thanks!

**Pull request merged**